### PR TITLE
Implemented requirements

### PR DESCRIPTION
**Pull request by selfbd**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -64,15 +64,21 @@ send me a DM to check your pull request
...
-    //2
-    //3
+    float value;
+    std::string name;
+
+    T(float v, const char* n)
+    {
+        value = v;
+        name = n;
+    }
```
Comment (matkatmusic) [inline main.cpp:74]: Cool.  you could have also used the member initializer list to init these members.

File: main.cpp [hunk 2]
```diff
@@ -64,15 +64,21 @@ send me a DM to check your pull request
...
+    }            
 };
 
-struct <#structName1#>                                //4
+struct MinHeight
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b)
     {
+        if (a == nullptr || b == nullptr) return nullptr;
```
Comment (matkatmusic) [inline main.cpp:81]: This works.  
Most people do this:
```
if( a != nullptr && b != nullptr )
{
    //the original comparison
}
return nullptr;
```

File: main.cpp [hunk 3]
```diff
@@ -81,47 +87,80 @@ struct <#structName1#>                                //4
...
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
-    
-    U <#name3#>;
+    std::cout << std::endl;
+    MinHeight f;
+    auto* smaller = f.compare(&person1, &person2);
+    if (smaller == nullptr)
+    {
+        std::cout << "Compare result is null" << std::endl;
+        return 0.f;
```
Comment (matkatmusic) [inline main.cpp:147]: - [ ] Don't return yet.   you'll skip all of the stuff that happens after this `if/else`

File: main.cpp (hunk 1 above)
Comment (selfbd) [reply main.cpp:74]: Changed constructor to use initializer list

File: main.cpp (hunk 2 above)
Comment (selfbd) [reply main.cpp:81]: Modified as suggested to simplify if block and nullptr checks

File: main.cpp (hunk 3 above)
Comment (selfbd) [reply main.cpp:147]: Implemented fix to avoid early exit if nullptr found

**Pull request merged**